### ISSUE TITLE
Having download Error is there any fix to this? I have space on my disk drive but it says no space. Please don't say there is no space on my device when there is 264 gigs still free on my disk. is there a better answer?

### ISSUE THREAD

**Issue by kstyle6228** [Franiac/TwitchLeecher#599]
Download task has been started!

TWITCH LEECHER INFO
--------------------------------------------------------------------------------------------
Version: 1.8.4

VOD INFO
--------------------------------------------------------------------------------------------
VOD ID: 930158811
Selected Quality: 1920x1080@60fps (Source)
Download Url: https://www.twitch.tv/videos/930158811
Crop Start: Yes (02:05:00)
Crop End: No

OUTPUT INFO
--------------------------------------------------------------------------------------------
Disable Conversion: No
Output File: D:\Adobe Projects\VODS\20210227_930158811_Just Chatting.mp4
FFMPEG Path: D:\Twitch Leecher\ffmpeg.exe
Temporary Download Folder: C:\Users\markp\AppData\Local\Temp\TL_5b168fdf-f125-46b7-bf10-20d4c9553633

ACCESS INFO
--------------------------------------------------------------------------------------------
Token: %7B%22authorization%22%3A%7B%22forbidden%22%3Afalse%2C%22reason%22%3A%22%22%7D%2C%22chansub%22%3A%7B%22restricted_bitrates%22%3A%5B%5D%7D%2C%22device_id%22%3Anull%2C%22expires%22%3A1619535084%2C%22https_required%22%3Atrue%2C%22privileged%22%3Afalse%2C%22user_id%22%3Anull%2C%22version%22%3A2%2C%22vod_id%22%3A930158811%7D
Signature: b5a17fa192ab4faa59ffbfa7984ba481387b0965
Sub-Only: No
Privileged: No

Creating temporary download directory 'C:\Users\markp\AppData\Local\Temp\TL_5b168fdf-f125-46b7-bf10-20d4c9553633'... done!

Retrieving m3u8 playlist urls for all VOD qualities... done!
https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/chunked/index-muted-H9DBT19WVG.m3u8
https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/720p60/index-muted-H9DBT19WVG.m3u8
https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/480p30/index-muted-H9DBT19WVG.m3u8
https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/audio_only/index-muted-H9DBT19WVG.m3u8
https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/360p30/index-muted-H9DBT19WVG.m3u8
https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/160p30/index-muted-H9DBT19WVG.m3u8

Playlist url for selected quality 1920x1080@60fps (Source) is https://dqrpb9wgowsf5.cloudfront.net/5a0e4eb0d65e21d9f1ba_dalphine_40776871245_1614407366/chunked/index-muted-H9DBT19WVG.m3u8

Retrieving playlist... done!
Parsing playlist... done!
Number of video chunks: 2342

Starting parallel video chunk download
Number of video chunks to download: 1593
Maximum connection count: 10

Parallel video chunk download is running...

Starting temporary download folder cleanup!
Deleting directory 'C:\Users\markp\AppData\Local\Temp\TL_5b168fdf-f125-46b7-bf10-20d4c9553633'... done!

Download task ended with an error!

System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`2 body)
   at TwitchLeecher.Services.Services.TwitchService.DownloadParts(Action`1 log, Action`1 setStatus, Action`1 setProgress, VodPlaylist vodPlaylist, CancellationToken cancellationToken)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass50_0.<StartQueuedDownloadIfExists>b__2()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`2 body)
   at TwitchLeecher.Services.Services.TwitchService.DownloadParts(Action`1 log, Action`1 setStatus, Action`1 setProgress, VodPlaylist vodPlaylist, CancellationToken cancellationToken)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass50_0.<StartQueuedDownloadIfExists>b__2()
   at System.Threading.Tasks.Task.Execute()
---> (Inner Exception #0) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #1) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #2) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #3) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #4) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #5) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #6) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #7) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---

---> (Inner Exception #8) System.IO.IOException: There is not enough space on the disk.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.File.InternalWriteAllBytes(String path, Byte[] bytes, Boolean checkHost)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass56_0.<DownloadParts>b__0(VodPlaylistPart part, ParallelLoopState loopState)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>)<---
<---

**Comment by Franiac** [comment]
Nope, you don't have space, my software is not lying 😄 

Maybe you have space in your download folder, but not in the temporary folder. Check your preferences and BOTH folders. Presumably your temporary folder is still set to C:\

**Comment by kstyle6228** [comment]
> Nope, you don't have space, my software is not lying 😄
> 
> Maybe you have space in your download folder, but not in the temporary folder. Check your preferences and BOTH folders. Presumably your temporary folder is still set to C:\

so I checked the C: temp folder and erased what was in it which wasn't much. and all my games are in my D: drive so I tried to clean up what I could and now I have 300 gigs free but can't download into it still. There is nothing more I can do @Franiac ?

**Comment by Franiac** [comment]
It's not about the temporary folder of Windows on drive C:\

There are two preferences in TL, the "Temporary Folder" and "Download Folder". To what drives are those set to?

**Comment by kstyle6228** [comment]
So I think I fixed it by moving the Temp and TMP folder to my game drive which is D:\ and did a disk cleanup which had 52 gigs on the temp folder which it took several attempts to erase all the temp files. After I started downloading VODs and it worked. I also moved the download folder to the D:\ drive so hope this will stay working as is. Thanks. Hope whoever has this problem in the future can do exactly what I did to fix.
